wgpu-hal/src/vulkan/command.rs

```diff
@@ -104,8 +111,10 @@ impl crate::CommandEncoder for super::CommandEncoder {
     }
 
     unsafe fn discard_encoding(&mut self) {
-        self.discarded.push(self.active);
-        self.active = vk::CommandBuffer::null();
+        if self.active != vk::CommandBuffer::null() {
+            self.discarded.push(self.active);
+            self.active = vk::CommandBuffer::null();
+        }
```

Comment:
Contrary to the original comments in #5255, it would probably be best to assert here that `self.active` is not null. If `wgpu-core` is calling `discard_encoding` multiple times, that is a bug in `wgpu-core`. The rationale is explained in #5566.